homeassistant/components/monarchmoney/config_flow.py

```diff
@@ -0,0 +1,145 @@
+"""Config flow for Monarch Money integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from monarchmoney import LoginFailedException, MonarchMoney, RequireMFAException
+from monarchmoney.monarchmoney import SESSION_FILE
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, CONF_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import CONF_MFA_CODE, DOMAIN, LOGGER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.EMAIL,
+            ),
+        ),
+        vol.Required(CONF_PASSWORD): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.PASSWORD,
+            ),
+        ),
+    }
+)
+
+STEP_MFA_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_MFA_CODE): str,
+    }
+)
+
+
+async def validate_login(
+    hass: HomeAssistant,
+    data: dict[str, Any],
+    email: str | None = None,
+    password: str | None = None,
+) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user. Upon success a session will be saved
+    """
+
+    if not email:
+        email = data[CONF_EMAIL]
+    if not password:
+        password = data[CONF_PASSWORD]
+    monarch_client = MonarchMoney()
+    if CONF_MFA_CODE in data:
+        mfa_code = data[CONF_MFA_CODE]
+        try:
+            LOGGER.debug("Attempting to authenticate with MFA code")
+            await monarch_client.multi_factor_authenticate(email, password, mfa_code)
+        except KeyError:
+            # A bug in the backing lib that I don't control throws a KeyError if the MFA code is wrong
+            LOGGER.debug("Bad MFA Code")
+            raise BadMFA from None
```

Comment:
from None?

Reply:
Ruff threw an error... 

```
B904 Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling
```

I'll just raise if rom the exception